packages/remix-server-runtime/server.ts

```diff
@@ -788,10 +858,125 @@ function unwrapResponse(response: Response) {
     : response.text();
 }
 
-function getSingleFetchRedirect(response: Response): SingleFetchRedirectResult {
+function getResponseStub(status?: number) {
+  let headers = new Headers();
+  let operations: ResponseStubOperation[] = [];
+  let headersProxy = new Proxy(headers, {
+    get(target, prop, receiver) {
+      if (prop === "set" || prop === "append" || prop === "delete") {
+        return (name: string, value: string) => {
+          operations.push([prop, name, value]);
+          Reflect.apply(target[prop], target, [name, value]);
+        };
+      }
+      return Reflect.get(target, prop, receiver);
+    },
+  });
+  return {
+    status,
+    headers: headersProxy,
+    [ResponseStubOperationsSymbol]: operations,
+  };
+}
+
+function getSingleFetchDataStrategy(
+  responseStubs: ReturnType<typeof getResponseStubs>
+) {
+  return async ({ request, matches }: DataStrategyFunctionArgs) => {
+    let results = await Promise.all(
+      matches.map(async (match) => {
+        let responseStub: ResponseStub | undefined;
+        if (request.method !== "GET") {
+          responseStub = responseStubs[ResponseStubActionSymbol];
+          if (!responseStub) {
+            responseStub = getResponseStub();
+            responseStubs[ResponseStubActionSymbol] = responseStub;
+          }
+        } else {
+          responseStub = responseStubs[match.route.id];
+          if (!responseStub) {
+            responseStub = getResponseStub();
+            responseStubs[match.route.id] = responseStub;
+          }
+        }
+
+        let result = await match.resolve(async (handler) => {
+          let data = await handler({ response: responseStub });
+          return { type: "data", result: data };
+        });
+
+        // Transfer raw Response status/headers to responseStubs
+        if (isResponse(result.result)) {
+          if (responseStub.status == null) {
+            responseStub.status = result.result.status;
+          }
+          for (let [k, v] of result.result.headers) {
+            if (k !== "Set-Cookie") {
+              responseStub.headers.set(k, v);
+            }
+          }
+
+          // Unsure why this is complaining?  It's fine in VSCode but fails with tsc...
+          // @ts-expect-error
+          for (let v of result.result.headers.getSetCookie()) {
```

Comment:
Confirming this is fine now that we use `undici`?  I know this was one of the areas we deviated from the spec and didn't yet implement this method